# Exercise Solution

In [1]:
from sklearn.datasets import load_iris
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
import numpy as np

In [2]:
iris = load_iris()

In [4]:
l_scores = cross_val_score(LogisticRegression(), iris.data, iris.target,cv=7)
l_scores  # Logistic Regression Scores

c:\users\shaon\appdata\local\programs\python\python37\lib\site-packages\sklearn\linear_model\_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)
c:\users\shaon\appdata\local\programs\python\python37\lib\site-packages\sklearn\linear_model\_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scik

array([0.95454545, 1.        , 0.90909091, 0.95238095, 0.95238095,
       1.        , 1.        ])

In [5]:
np.average(l_scores)   # LogisticRegression Average

0.9669140383426098

In [7]:
d_scores = cross_val_score(DecisionTreeClassifier(), iris.data, iris.target,cv=7)
d_scores               # Decision Tree scores

array([0.95454545, 1.        , 0.90909091, 0.85714286, 0.95238095,
       0.95238095, 1.        ])

In [8]:
np.average(d_scores)    # DecisionTree Average

0.9465058750773038

In [11]:
s_scores = cross_val_score(SVC(), iris.data, iris.target,cv=7) # SVM Scores
s_scores

array([0.95454545, 1.        , 1.        , 0.95238095, 0.95238095,
       0.95238095, 1.        ])

In [12]:
np.average(s_scores)   # SVMaverage

0.9730983302411874

In [14]:
r_scores = cross_val_score(RandomForestClassifier(n_estimators=50), iris.data, iris.target,cv=7)
r_scores               # Random Forest Scores

array([0.95454545, 1.        , 0.90909091, 0.95238095, 0.85714286,
       0.95238095, 1.        ])

In [15]:
np.average(r_scores)    # RandomForest average

0.9465058750773035

#### Best score so far is 0.9730983 which is generated by SVM model